src/dj_notebook/config_helper.py

```diff
@@ -0,0 +1,124 @@
+import ast
+import os
+from pathlib import Path
+from typing import Generator, Tuple
+
+from dotenv import load_dotenv
+
+
+# taken from dotenv, which declares a similar type (but it doesn't look public...)
+# review note: the | syntax is new in python 3.10. If older pythons are generally being supported here, this should be
+# rewritten as Union[str, os.PathLike[str]]
+StrPath = str | os.PathLike[str]
+
+
+def setdefault_calls(module_path: Path) -> Generator[ast.Call, None, None]:
+    """Yields all calls to `os.environ.setdefault` within a module."""
+    with open(module_path, "r") as module_src:
+        parsed_module = ast.parse(module_src.read())
+    environ_id = None
+    for node in ast.walk(parsed_module):
+        if isinstance(node, ast.ImportFrom) and node.module == "os":
+            for name in node.names:
+                if isinstance(name, ast.alias) and name.name == "environ":
+                    environ_id = name.asname if name.asname is not None else name.name
+        if (
```

Comment:
Curiously, this code block helps me understand what `ast` does so much better than any documentation I've read on the topic.